lib/Optimizer/CodeGen/LowerToQIRProfile.cpp

```diff
@@ -65,10 +65,7 @@ namespace {
 struct FunctionAnalysisData {
   std::size_t nQubits = 0;
   std::size_t nResults = 0;
-  // Use std::map to keep these sorted in ascending order.
```

Comment:
I would keep the rationale for using std::map comment here. LLVM programmers are conditioned by reflex that using std::map is probably an error, so the comment is meant to reassure that we know what we are doing here.

Caveat: assuming that a sorted order is still important, of course. And if it isn't then we should use one of the LLVM ADT maps here instead.

Reply:
I think it is no longer _required_, but using `DenseMap` here will produce unsorted instructions in the QIR, which seems undesirable to me. I will add a comment stating as much.

Reply:
Added comment in f1e6886e.